### PR TITLE
Feature/publish to testpypi

### PR DESCRIPTION
**Pull request by JJdeVries**

### CODE REVIEW

File: .github/workflows/release_deploy_pypi.yml [hunk 1]
```diff
@@ -1,8 +1,41 @@
...
+
+      - name: Setup python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.2.2
```
Comment (tjadevries) [inline .github/workflows/release_deploy_pypi.yml:30]: Why this version? @ viro we had a specific problem with this one, related to installation under windows so not that important for this workflow. But in general I think that it is worthwhile to update, as many fixes are made (https://github.com/python-poetry/poetry/blob/master/CHANGELOG.md)

File: .github/workflows/release_deploy_pypi.yml [hunk 2]
```diff
@@ -1,8 +1,41 @@
...
+    permissions:
+      id-token: write
+
     steps:
-      - name: "Temp"
-        run: echo "Temporary, too be implemented"
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup python 3.10
```
Comment (tjadevries) [inline .github/workflows/release_deploy_pypi.yml:22]: Why 3.10 and not 3.11?

**Review by tjadevries — CHANGES_REQUESTED** [review]

File: .github/workflows/release_deploy_pypi.yml [hunk 3]
```diff
@@ -1,8 +1,41 @@
...
+      url: https://pypi.org/p/application-settings
+
+    permissions:
+      id-token: write
+
     steps:
-      - name: "Temp"
-        run: echo "Temporary, too be implemented"
+      - name: Checkout repo
+        uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (JJdeVries) [reply .github/workflows/release_deploy_pypi.yml:20]: I don't think this workflow should really specify that? Essentially the checkout action runs for whatever branch we trigger the release. Essentially that will be 'master' but we should not enforce that in this workflow I think.

File: .github/workflows/release_deploy_pypi.yml [hunk 4]
```diff
@@ -1,8 +1,41 @@
...
+          virtualenvs-in-project: true
+
+      - name: Check lockfile
+        run: poetry lock --check
+
+      - name: Build dist
+        run: poetry build
+
+      - name: Publish package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@b7f401de30cb6434a1e19f805ff006643653240e
```
[the remark this answers is not in the document]
Comment (JJdeVries) [reply .github/workflows/release_deploy_pypi.yml:41]: Do we want that? The branch is named a bit awkward but I think we only really care about test pi. (although changing is as easy as changing the url to test.pypi.

File: .github/workflows/release_deploy_pypi.yml (hunk 1 above)
Comment (JJdeVries) [reply .github/workflows/release_deploy_pypi.yml:30]: I copied the version of the action from the other workflow files, I'll update those. I actually expected that the dependabot should check that?

File: .github/workflows/release_deploy_pypi.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tjadevries) [reply .github/workflows/release_deploy_pypi.yml:20]: Ah, now I understand which branch is checked out. Okay.

File: .github/workflows/release_deploy_pypi.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tjadevries) [reply .github/workflows/release_deploy_pypi.yml:41]: Okay

**Review by tjadevries — APPROVED** [review]

**Pull request merged**